src/client/renderer/r_state.c

```diff
@@ -917,6 +985,16 @@ void R_UseInterpolation(const vec_t lerp) {
 }
 
 /**
+ * @brief Uploads shell offset value to the currently loaded program.
+ */
+void R_UseShellOffset(const vec_t offset) {
+
+	if (r_state.active_program->UseShellOffset) {
+		r_state.active_program->UseShellOffset(offset);
+	}
+}
+
```

Comment:
I was grossed out by it too, and I was trying to think of a way to directly access the state of the shell program without causing too much trouble but I just couldn't do it elegantly. My first thought was just as yours, to hijack the Use function, but Use is called wayyyy before entities get managed and we can't call Use for each entity because it will skip it (it won't try to Use a program that is already Use'd) and seems kind of inefficient.